flink-ml-lib/src/test/java/org/apache/flink/ml/feature/BinarizerTest.java

```diff
@@ -0,0 +1,164 @@
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+/** Tests {@link Binarizer}. */
+public class BinarizerTest extends AbstractTestBase {
+
+    private StreamTableEnvironment tEnv;
+    private Table inputDataTable;
+
+    private static final List<Row> INPUT_DATA =
+            Arrays.asList(
+                    Row.of(
+                            1,
+                            Vectors.dense(1, 2),
+                            Vectors.sparse(17, new int[] {0, 3, 9}, new double[] {1.0, 2.0, 7.0})),
+                    Row.of(
+                            2,
+                            Vectors.dense(2, 1),
+                            Vectors.sparse(17, new int[] {0, 2, 14}, new double[] {5.0, 4.0, 1.0})),
+                    Row.of(
+                            3,
+                            Vectors.dense(5, 18),
+                            Vectors.sparse(
+                                    17, new int[] {0, 11, 12}, new double[] {2.0, 4.0, 4.0})));
+
+    private static final Double[] EXPECTED_VALUE_OUTPUT = new Double[] {0.0, 1.0, 1.0};
+
+    private static final List<Vector> EXPECTED_DENSE_OUTPUT =
+            Arrays.asList(
+                    new DenseVector(new double[] {0.0, 1.0}),
+                    new DenseVector(new double[] {1.0, 0.0}),
+                    new DenseVector(new double[] {1.0, 1.0}));
+
+    private static final List<Vector> EXPECTED_SPARSE_OUTPUT =
+            Arrays.asList(
+                    Vectors.sparse(17, new int[] {9}, new double[] {1.0}),
+                    Vectors.sparse(17, new int[] {0, 2}, new double[] {1.0, 1.0}),
+                    Vectors.sparse(17, new int[] {11, 12}, new double[] {1.0, 1.0}));
+
+    @Before
+    public void before() {
+        Configuration config = new Configuration();
+        config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+
+        env.setParallelism(4);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+
+        tEnv = StreamTableEnvironment.create(env);
+        DataStream<Row> dataStream = env.fromCollection(INPUT_DATA);
+        inputDataTable = tEnv.fromDataStream(dataStream).as("f0", "f1", "f2");
+    }
+
+    private void verifyOutputResult(Table output, String[] outputCols) throws Exception {
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) output).getTableEnvironment();
+        DataStream<Row> stream = tEnv.toDataStream(output);
+
+        List<Row> results = IteratorUtils.toList(stream.executeAndCollect());
+        List<Double> values = new ArrayList<>(results.size());
+        List<Vector> sparseVec = new ArrayList<>(results.size());
+        List<Vector> denseVec = new ArrayList<>(results.size());
```

Comment:
nit: Let's keep the naming rule consistent across these variables. For example, `doubleValues`, `denseVectorValues`, `sparseVectorValues`, or `numbers`, `denseVectors`, `sparseVectors`.